parser.py

```diff
@@ -162,9 +178,37 @@ def main():
         filename = filename = f'{dt.year}-{dt.month:02}-01-Tweet-Archive-{dt.year}-{dt.month:02}.md' # change to group by day or year or timestamp
         grouped_tweets[filename].append(md)
 
+    tweet_files = {}
+    for filename, md in grouped_tweets.items():
+        for t in md:
+            tweet_files[t.id] = filename
+
+    forward_refs = defaultdict(list)
+    for filename, md in grouped_tweets.items():
+        for t in md:
+            if t.reply_to_id and t.reply_to_id in tweet_files:
+                forward_refs[t.reply_to_id].append(t.id)
+
     # Write into files
     for filename, md in grouped_tweets.items():
-        md_string =  '\n\n----\n\n'.join(md)
+        md_string = ''
+        prev_id = ''
+        for t in md:
+            # got successors?
+            first_forward = True
+            for s in forward_refs[prev_id]:
+                if s != t.id:
+                    md_string += '\n\n' if first_forward else ''
+                    first_forward = False
+                    md_string += f' [↓]({tweet_files[s]}#s{s})'
+            if (not t.reply_to_id or t.reply_to_id != prev_id):
+                # not a reply to the last output tweet: write a horizontal rule
+                md_string += '\n\n----\n' if md_string else ''
+                if(t.reply_to_self):
+                    # a reply-to-self, but not a reply to the most recently output tweet: add a header
+                    md_string += f'\n[↑]({tweet_files[t.reply_to_id]}#s{t.reply_to_id})' if t.reply_to_id in tweet_files else '↑'
```

Comment:
This breaks the new html output, sadly, since it links to an md file.

Reply:
@twoscomplement I'm thinking of rewriting the HTML output to be inside parser.py, produced alongside the markdown. This would make fixing this issue trivial.